Automattic/jetpack
# Add transport helper functionality

#### Description
- Currently, Jetpack and the standalone Jetpack Backup plugin rely on the `backup` package for helper script installation, deletion, and clearing. With the introduction of Jetpack Protect and a third plugin that will require this functionality, we have decided to separate out and generalize all helper script related code (endpoints, cron, and other management) into a separate `transport-helper` package so that the tooling can be easily shared between the three reliant plugins and any others that may have the requirement in the future.
- The intention of this initial pass is to create a package that only contains the helper script specific code from the `backup` package, that will work in parallel until a time when we are confident that the code can be removed for the `backup` package and all reliant internal/external code has been updated to reflect the change so that we can confidently rely solely on the `transport-helper` package for these actions.

#### Changes proposed in this Pull Request:

- [x] Create a new `transport-helper` package (#25772)
- [x] Duplicate and generalize all helper script code to the `transport-helper` package, with the exception of the registered endpoint route `/backup-helper-script`
- [x] Update WPcom code reliant on the `backup` package for processing helper script requests to account for the new package (D87446-code)

#### Next Steps:
- Remove any/all helper script code from the `backup` package, and generalize the registered endpoint route to `/helper-script`
- Update any/all internal code reliant on the helper script code from the `backup` package accordingly
- Update any/all external code reliant on the helper script code from the `backup` package accordingly

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1202645981599451
- p1659461330589659-slack-CS8UYNPEE

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:





* With 87446-code now deployed, proceed with the following:
* Start up a Jurassic Ninja instance, only including the Jetpack Beta plugin and this Protect branch
* Add a Scan plan to the site
* Open up the RWDB and verify that the `auto-http` protocol cxn tests pass
* Queue a scan and verify that it succeeds
